pkg/trace/config/config.go

```diff
@@ -606,6 +605,15 @@ func (c *AgentConfig) AllFeatures() []string {
 	return feats
 }
 
+// ConfiguredPeerTags returns the set of peer tags that should be used
+// for aggregation based on the various config values and the base set of tags.
+func (c *AgentConfig) ConfiguredPeerTags() []string {
+	if !c.PeerTagsAggregation {
+		return nil
+	}
+	return preparePeerTags(append(basePeerTags, c.PeerTags...))
```

Comment:
Good call out. It looks like otelSpanToDDSpan was doing that before, so this shouldn't introduce new cost. I'm not sure if caching in the config is ideal or not, as it introduces dependencies on the other fields (what if a user changes a field?). I don't think it's likely that someone would change the config after use, but if they do it would result in weird behavior.

I'll see if I can write up a small benchmark and if we can cache this in the config or somewhere more sensible.

Reply:
yeah seems like it's safe to keep as-is for now and maybe leave a todo or make a ticket to look into it later? Don't want to grow this PR too much beyond what we have already :)